expression/aggregation/util.go

```diff
@@ -68,7 +69,22 @@ func calculateSum(sc *stmtctx.StatementContext, sum, v types.Datum) (data types.
 			data = types.NewDecimalDatum(d)
 		}
 	case types.KindMysqlDecimal:
-		data = types.CloneDatum(v)
+		dec := v.GetMysqlDecimal()
+		frac := retType.Decimal
+		if frac == types.UnspecifiedLength {
+			frac = mysql.MaxDecimalScale
+		}
+		if int(dec.GetDigitsFrac()) > frac {
+			to := new(types.MyDecimal)
+			err := dec.Round(to, frac, types.ModeHalfEven)
+			data = types.Datum{}
+			data.SetMysqlDecimal(to)
+			if err != nil {
+				return data, err
```

Comment:
Why put `return` after creating `data`? If `err != nil`, `to` is not even integral.

Reply:
Yes, we should return the data even though overflow occurred (the caller should handle it in a proper way if the err is overflow or truncated).